envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/clusters/EnvoyClustersFactory.kt

```diff
@@ -283,15 +283,11 @@ class EnvoyClustersFactory(
     ): Boolean {
         val trafficSplitting = properties.loadBalancing.trafficSplitting
         val trafficSplitEnabled = trafficSplitting.weightsByService.containsKey(serviceName)
-        return trafficSplitEnabled && hasEndpointsInZone(clusterLoadAssignment, trafficSplitting)
+        val allowed = clusterLoadAssignment != null &&
+            properties.loadBalancing.trafficSplitting.zonesAllowingTrafficSplitting.contains(currentZone)
+        return trafficSplitEnabled && allowed
     }
 
-    private fun hasEndpointsInZone(
-        clusterLoadAssignment: ClusterLoadAssignment?,
-        trafficSplitting: TrafficSplittingProperties
-    ) = clusterLoadAssignment?.endpointsList
-        ?.any { e -> trafficSplitting.zoneName == e.locality.zone && e.lbEndpointsCount > 0 } ?: false
```

Comment:
Why we no longer check if there are any endpoints?

Reply:
yea I think I had some justification when I was deleting it, but now don't recall why, so returned it back